src/Autofac.Integration.Mef/Autofac.Integration.Mef.csproj

```diff
@@ -1,102 +1,117 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Autofac" Version="6.0.0" />
-    <PackageReference Include="Microsoft.CodeAnalysis.Analyzers" Version="3.3.2">
+    <PackageReference Include="Autofac" Version="6.4.0" />
+    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.1.1" Condition="Exists('$(MSBuildThisFileDirectory)../../.git')">
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
-    <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="5.0.3">
-      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
-      <PrivateAssets>all</PrivateAssets>
-    </PackageReference>
-    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" Condition="Exists('$(MSBuildThisFileDirectory)../../.git')">
-      <PrivateAssets>All</PrivateAssets>
-    </PackageReference>
-    <PackageReference Include="StyleCop.Analyzers" Version="1.2.0-beta.354">
+    <PackageReference Include="StyleCop.Analyzers" Version="1.2.0-beta.406">
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
     <PackageReference Include="System.ComponentModel.Composition" Version="4.5.0" />
   </ItemGroup>
 
   <ItemGroup>
-    <Compile Update="ContractBasedServiceResources.Designer.cs">
-      <AutoGen>True</AutoGen>
-      <DesignTime>True</DesignTime>
-      <DependentUpon>ContractBasedServiceResources.resx</DependentUpon>
-    </Compile>
-  </ItemGroup>
-
-  <ItemGroup>
-    <Compile Update="LazyWithMetadataRegistrationSourceResources.Designer.cs">
-      <AutoGen>True</AutoGen>
-      <DesignTime>True</DesignTime>
-      <DependentUpon>LazyWithMetadataRegistrationSourceResources.resx</DependentUpon>
-    </Compile>
-  </ItemGroup>
-
-  <ItemGroup>
-    <Compile Update="StronglyTypedMetadataRegistrationSourceResources.Designer.cs">
-      <AutoGen>True</AutoGen>
-      <DesignTime>True</DesignTime>
-      <DependentUpon>StronglyTypedMetadataRegistrationSourceResources.resx</DependentUpon>
-    </Compile>
-  </ItemGroup>
+    <!--
+      Magic embedded resource incantation based on https://github.com/dotnet/msbuild/issues/4751
 
-  <ItemGroup>
-    <Compile Update="RegistrationExtensionsResources.Designer.cs">
-      <AutoGen>True</AutoGen>
-      <DesignTime>True</DesignTime>
-      <DependentUpon>RegistrationExtensionsResources.resx</DependentUpon>
-    </Compile>
-  </ItemGroup>
-
-  <ItemGroup>
-    <Compile Update="Util\ReflectionExtensionsResources.Designer.cs">
-      <AutoGen>True</AutoGen>
-      <DesignTime>True</DesignTime>
-      <DependentUpon>RegistrationExtensionsResources.resx</DependentUpon>
-    </Compile>
+      The EmbeddedResource entry seems to add the Designer files to Compile, so
+      if you don't first remove them from Compile you get warnings about
+      double-including source.
+    -->
+    <Compile Remove="**/*.Designer.cs" />
+    <EmbeddedResource Update="ContractBasedServiceResources.resx">
```

Comment:
If these files are now generated by a build task that functions correctly during dotnet build as well as in VS, should the *.Designer.cs files be removed from disk and added to a .gitignore?  Or do they still need to be in source?

Reply:
It appears that we could remove them from source since they'll be brought back on build. It implies you have to build after you check out, but... don't folks do that anyway? I'll update the PR here to remove them.